### ISSUE TITLE
TypeError

### ISSUE THREAD

**Issue by grid-rider** [IFCjs/hello-world#1]
Hey agviegas thankyou for creating such a great tutorial for the IFC.js-three.js beginners. 

Yesterday I attempted using the hello-world tutorial to build my first ifc viewer and did so by using the hello-world html and app.js. I then downloaded the web-ifc.wasm file and bundled the projekt using **webpack**. After opening the site the 3d scene appears, but the scene is unable to add the ifc model. I recompiled multiple times and switched between local and external ifc files and when the ifc  files are attempted to be loaded (by app.js) I always receive the error:
TypeError: Cannot read property 'rawConstructor' of undefined
    at pe (bundle_3d_viewer.js:2)
    at web-ifc.wasm:0x45b2d
    at X (web-ifc.wasm:0x301c2)
    at r.___wasm_call_ctors (bundle_3d_viewer.js:2)
    at func (bundle_3d_viewer.js:2)
    at _t (bundle_3d_viewer.js:2)
    at pt (bundle_3d_viewer.js:2)
    at s (bundle_3d_viewer.js:2)
    at Ui (bundle_3d_viewer.js:2)
    at t (bundle_3d_viewer.js:2)
(anonymous) @ bundle_3d_viewer.js:2
async function (async)
(anonymous) @ bundle_3d_viewer.js:2
(anonymous) @ bundle_3d_viewer.js:2
load (async)
load @ bundle_3d_viewer.js:2
load @ bundle_3d_viewer.js:2
(anonymous) @ bundle_3d_viewer.js:2

<img width="1013" alt="Screenshot 2021-07-06 110643" src="https://user-images.githubusercontent.com/62141711/124574015-592d0180-de4a-11eb-885c-fa22a9758b6c.png">

my app.js that I used for bundling and the corresponding webpack.config.js is also added below to give more context.

app.js
```
import {
  AmbientLight,
  AxesHelper,
  Color,
  DirectionalLight,
  GridHelper,
  PerspectiveCamera,
  Scene,
  Vector3,
  WebGLRenderer,
} from "three";
import { OrbitControls } from "three/examples/jsm/controls/OrbitControls";
import { IFCLoader } from "three/examples/jsm/loaders/IFCLoader";

//Creates the Three.js scene
const scene = new Scene();

//Object to store the size of the viewport
const size = {
  width: window.innerWidth,
  height: window.innerHeight,
};

//Creates the camera (point of view of the user)
const camera = new PerspectiveCamera(75, size.width / size.height);
camera.position.z = 15;
camera.position.y = 13;
camera.position.x = 8;

//Creates the lights of the scene
const lightColor = 0xffffff;

const ambientLight = new AmbientLight(lightColor, 0.5);
scene.add(ambientLight);

const directionalLight = new DirectionalLight(lightColor, 1);
directionalLight.position.set(0, 10, 0);
directionalLight.target.position.set(-5, 0, 0);
scene.add(directionalLight);
scene.add(directionalLight.target);

//Sets up the renderer, fetching the canvas of the HTML
const threeCanvas = document.getElementById("three-canvas");
const renderer = new WebGLRenderer({ canvas: threeCanvas, alpha: true });
renderer.setSize(size.width, size.height);
renderer.setPixelRatio(Math.min(window.devicePixelRatio, 2));

//Creates grids and axes in the scene
const grid = new GridHelper(50, 30);
scene.add(grid);

const axes = new AxesHelper();
axes.material.depthTest = false;
axes.renderOrder = 1;
scene.add(axes);

//Creates the orbit controls (to navigate the scene)
const controls = new OrbitControls(camera, threeCanvas);
controls.enableDamping = true;
controls.target.set(-2, 0, 0);

//Animation loop
const animate = () => {
  controls.update();
  renderer.render(scene, camera);
  requestAnimationFrame(animate);
};

animate();

//Adjust the viewport to the size of the browser
window.addEventListener("resize", () => {
  (size.width = window.innerWidth), (size.height = window.innerHeight);
  camera.aspect = size.width / size.height;
  camera.updateProjectionMatrix();
  renderer.setSize(size.width, size.height);
});

//Sets up the IFC loading

function loadIFC(){}
const ifcLoader = new IFCLoader();
ifcLoader.setWasmPath("../");
const url = "https://raw.githubusercontent.com/IFCjs/test-ifc-files/main/Revit/TESTED_Simple_project_01.ifc";

window.addEventListener('DOMContentLoaded', (event) => {
  ifcLoader.load(url, (geometry) => scene.add(geometry));
});
```
webpack.config.js
```
const path = require('path');
module.exports = {
    entry: "./src/public/js/app.js",
    output: {
        filename: 'bundle_3d_viewer.js',
        path: path.join(__dirname,"/src/public/dist")
    },

    devServer: {
        contentBase: path.join(__dirname,'/src/public'),
        port:9000
    }
};
```
Here is the viewer.html, which is essentially the same as the tutorial, but with different paths.
```
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <link rel="stylesheet" href="../css/platform.css" />
    <title>Document</title>
  </head>
  <body>
    <input type="file" name="load" id="file-input" />
    <canvas id="three-canvas"></canvas>
    <script src="../dist/bundle_3d_viewer.js"></script>
  </body>
</html>
```
I suspect it is an issue with the wasm file, but it could also be the app.js itself or my bundelling technique. 

Do you have an idea how I could fix it?

**Comment by agviegas** [comment]
Hey, I'm not that familiar with webpack. Generally you have to make sure of 2 things: 1. That you are statically serving the WASM, and 2. that the WASM has the correct version. If you picked the WASM from three and the Loader from three, this shouldn't be your problem.

**Comment by agviegas** [comment]
As far as I recall, you have to configure webpack to serve each type of file. Did you configure webpack to handle the wasm (serve it statically)?

**Comment by grid-rider** [comment]
Thank you for the great and quick feedback. 
I was not able to get it to work, but after switching to rollup it worked perfectly and is now loading successfully!

I am now currently working on trying to change the Materials of the different elements of the ifc file and wanted to firstly  ask if  that is even possible. If yes can you do this by accessing and **changing** the ifc elements through three.js methods? And how would the approach to this be ? (through expressid,globalid,etc..).

**Comment by agviegas** [comment]
Great! Yes, you can do it, but you need to use the API we provide. If you are using Three.js, you'll have to wait until we make the PR with the last changes (we will submit it today). In addition, we are working hard to bring everything to the docs. Hopefully they will be done before the end of next week. 😊 In the meantime, feel free to get into the open Discord group and ask everything you need there!

**Comment by grid-rider** [comment]
Ok that sounds great and it's amazing to hear about the continues work you guys are putting in.
I just wanted to quickly ask if you have already published the three.js api, because if yes I currently can't find it😊 and I am eager to continue.

**Comment by agviegas** [comment]
What do you mean by three.js api? Maybe [this](https://github.com/IFCjs/web-ifc-three)?

**Comment by grid-rider** [comment]
Sry. I meant the PR with the last changes that you described in the previous comment, that I should wait for.

**Comment by agviegas** [comment]
[Here](https://github.com/mrdoob/three.js/pull/22113) it is! Pending to merge. And also we are putting some effort in the [docs ](https://ifcjs.github.io/info/ )these weeks.

**Comment by grid-rider** [comment]
Ok great thanks.
Is it enough if I now run npm update to get the newest web-ifc-three components, because I am currently attempting to use the ifcmodel.mesh property when adding to the array and it is still throwing an error (to be specific this error: `THREE.Object3D.add: object not an instance of THREE.Object3D.`)?

**Comment by agviegas** [comment]
Nope, you have to wait until the next Three.js release. You can use [the loader](https://github.com/IFCjs/web-ifc-three) directly though. Clone the dist folder or get it with `npm install web-ifc-three`.
